battleship-service/src/test/java/dreamteam/battleship/service/RegistrationTest.java

```diff
@@ -0,0 +1,60 @@
+package dreamteam.battleship.service;
+
+import org.springframework.mock.web.MockHttpSession;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Created by ehsan on 18.07.16.
+ */
+
+public class RegistrationTest {
+
+    @DataProvider
+    protected Object[][] generatedKeys(){
+        KeyGenerator generator = new KeyGenerator();
+        Set<String> temp = new HashSet<String>();
+        return new Object[][]{
+                {temp.add(generator.generate())},
+                {temp.add(generator.generate())},
+                {temp.add(generator.generate())},
+                {temp.add(generator.generate())},
+                {temp.add(generator.generate())},
+                {temp.add(generator.generate())},
+                {temp.add(generator.generate())},
+        };
+    }
+
+    @Test(dataProvider = "generatedKeys")
```

Comment:
I don't understand this test. DataProvider return String. Probably the name of method is misleading.